src/ipc_cache.c

```diff
@@ -144,6 +144,8 @@ umfIpcOpenedCacheCreate(ipc_opened_cache_eviction_cb_t eviction_cb) {
 
 void umfIpcOpenedCacheDestroy(ipc_opened_cache_handle_t cache) {
     ipc_opened_cache_entry_t *entry, *tmp;
+    utils_mutex_lock(&(cache->global->cache_lock));
```

Comment:
I do not think that lock is needed here because destructor is always called at the end and there should not be any concurrent threads. I think this Coverity issue should be marked as a false-positive.

Reply:
@vinser52 do we need to set cur_size here? Maybe we could remove this line so Coverity would not report any error

cache->global->cur_size -= 1;

Reply:
ahh, wait, this is not a global cache destructor, this function is called every time a particular pool/ipc_handler is destroyed. If we have two or more ipc_handlers that are destroyed in parallel then this function is called concurrently. So looks like we need the lock here. We need a test for such flow.

Give me some time to add such a test in `ipcFixtures.hpp`